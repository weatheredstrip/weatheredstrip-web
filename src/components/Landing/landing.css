.swing-in-top-fwd {
	-webkit-animation: swing-in-top-fwd 0.5s cubic-bezier(0.175, 0.885, 0.320, 1.275) both;
	        animation: swing-in-top-fwd 0.5s cubic-bezier(0.175, 0.885, 0.320, 1.275) both;
}

.swing-in-top-fwd:before {
    content: '""';
    width: 3;
    height: 100%;
    position: 'absolute';
    top: 0;
    left: 0;
    background-color: #1070CA;
  }

/* ----------------------------------------------
 * Generated by Animista on 2020-5-20 14:10:9
 * Licensed under FreeBSD License.
 * See http://animista.net/license for more info. 
 * w: http://animista.net, t: @cssanimista
 * ---------------------------------------------- */

/**
 * ----------------------------------------
 * animation swing-in-top-fwd
 * ----------------------------------------
 */
@-webkit-keyframes swing-in-top-fwd {
  0% {
    -webkit-transform: rotateX(-100deg);
            transform: rotateX(-100deg);
    -webkit-transform-origin: top;
            transform-origin: top;
    opacity: 0;
  }
  100% {
    -webkit-transform: rotateX(0deg);
            transform: rotateX(0deg);
    -webkit-transform-origin: top;
            transform-origin: top;
    opacity: 1;
  }
}
@keyframes swing-in-top-fwd {
  0% {
    -webkit-transform: rotateX(-100deg);
            transform: rotateX(-100deg);
    -webkit-transform-origin: top;
            transform-origin: top;
    opacity: 0;
  }
  100% {
    -webkit-transform: rotateX(0deg);
            transform: rotateX(0deg);
    -webkit-transform-origin: top;
            transform-origin: top;
    opacity: 1;
  }
}

