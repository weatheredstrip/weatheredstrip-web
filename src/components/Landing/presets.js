import React, { Component } from 'react'
import { withFirebase } from ''

const Presets = () => {

}

export default Presets