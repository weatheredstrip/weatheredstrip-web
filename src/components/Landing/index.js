import React, { Component, useState } from 'react'
import { Alert, Pane, Text } from 'evergreen-ui'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faSearch } from '@fortawesome/free-solid-svg-icons'
import { withRouter } from 'react-router-dom'
import './landing.css'

import { AuthUserContext } from '../Session';
import Navigation from '../Navigation'
import { PrivacyPolicyDialog, ToSDialog } from '../Policies'

import * as ROUTES from '../../constants/routes'

import logo from '../logo.svg'

const year = new Date();

const NewAccountAlert = () => {
  const [display, setDisplay] = useState('flex')
  return (
    <Pane
      position="absolute"
      top={20}
      right="50%"
      zIndex={1}
    >
      <AuthUserContext.Consumer>
        {authUser => !authUser && (
            <Alert
              hasTrim={true}
              intent="none"
              position="relative"
              right="-50%"
              display={display}
              appearance="card"
              title="You can now create an account!"
              maxWidth={360}
              minWidth={200}
              hasIcon={false}
              isRemoveable={true}
              onRemove={() => setDisplay('none')}
            />
        )}
      </AuthUserContext.Consumer>
    </Pane>
)
}

const LandingPage = ({privacy=false, terms=false}) => {
  const [isPrivacyShown, setPrivacy] = useState(privacy)
  const [isTermsShown, setTerms] = useState(terms)
  return (
    <>
      <Navigation position="absolute" top="20px" right="20px" appearance="default"/>
      <NewAccountAlert />
      <div className="landing-page">
        <div className="landing-content">
          <div className="logo-title">
            <img className="landing-logo" src={logo} alt="App Logo" />
            <div className="landing-title">Weathered Strip</div>
          </div>
          <SearchForm />
        </div>
        <Pane
          position="absolute"
          bottom={20}
        >
          <Text
            color="white"
            size={300}
          >
            By Greg Hamel &copy; {year.getUTCFullYear()}
          </Text>
        </Pane>
      </div>
      <PrivacyPolicyDialog
        isShown={isPrivacyShown}
        onCloseComplete={() => setPrivacy(false)}
      />
      <ToSDialog
        isShown={isTermsShown}
        onCloseComplete={() => setTerms(false)}
      />
    </> 
  )
}

const INITIAL_STATE = {
  search: '',
}

class SearchFormBase extends Component {
  constructor(props) {
    super(props)
    this.state = { ...INITIAL_STATE }
  }

  onSubmit = event => {
    event.preventDefault()
    const { search } = this.state
    const request = {
      pathname: ROUTES.REQUEST,
      search: '?stations=' + search,
    }

    this.setState({ ...INITIAL_STATE })
    this.props.history.push(request)
  }

  onChange = event => {
    this.setState({ [event.target.name]: event.target.value.toUpperCase() })
  }

  render() {
    return (
      <form onSubmit={this.onSubmit} className="landing-search">
        <input
          className="landing-search-input"
          autoComplete="off"
          autoCorrect="off"
          autoCapitalize="off"
          spellCheck="false"
          name="search"
          placeholder="ICAO or IATA codes"
          value={this.state.search}
          onChange={this.onChange}
        />
        <button className="landing-search-button" type="submit">
          <FontAwesomeIcon
            className="landing-search-button-icon"
            icon={faSearch}
            size="2x"
            color="#282c34"
          />
        </button>
      </form>
    )
  }
}

const SearchForm = withRouter(SearchFormBase)

export default LandingPage
