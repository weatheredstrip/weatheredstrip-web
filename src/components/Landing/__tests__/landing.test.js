import React from 'react'
import { render, cleanup, fireEvent, screen } from '@testing-library/react'
import { 
  BrowserRouter as Router
} from 'react-router-dom'
import Landing from '../index'
import '@testing-library/jest-dom/extend-expect'
afterEach(cleanup)

it('landing renders properly', () => {
  const { container } = render(<Router><Landing /></Router>)

  expect(container).toMatchSnapshot()
})

it('has navigation button', () => {
  const { container } = render(<Router><Landing/></Router>)
  const landing = container.firstChild
  expect(landing.firstChild.nodeName).toEqual("BUTTON")
})

it('opens navigation when button is clicked', () => {
  const { container } = render(<Router><Landing /></Router>)

  const landing = container.firstChild

  const button = landing.firstChild

  fireEvent(
    button,
    new MouseEvent('click', {
      bubbles: true,
      cancelable: false,
    })
  )

  const menu = screen.getByRole("menu");
  expect(menu).not.toBeNull();
})