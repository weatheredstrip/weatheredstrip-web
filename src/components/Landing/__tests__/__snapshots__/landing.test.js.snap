// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`landing renders properly 1`] = `
<div
  class="css-nil ðŸ“¦pst_absolute ðŸ“¦rgt_20px ðŸ“¦top_20px ðŸ“¦box-szg_border-box"
>
  <button
    class="css-69cngj ðŸ“¦fnt-fam_b77syt ðŸ“¦mt_0px ðŸ“¦fnt-sze_12px ðŸ“¦f-wght_500 ðŸ“¦ln-ht_32px ðŸ“¦ltr-spc_0 ðŸ“¦btrr_3px ðŸ“¦bbrr_3px ðŸ“¦btlr_3px ðŸ“¦bblr_3px ðŸ“¦pt_0px ðŸ“¦pb_0px ðŸ“¦pr_0px ðŸ“¦pl_0px ðŸ“¦ml_0px ðŸ“¦mr_0px ðŸ“¦mb_0px ðŸ“¦h_32px ðŸ“¦pst_relative ðŸ“¦dspl_flex ðŸ“¦algn-itms_center ðŸ“¦flx-wrap_nowrap ðŸ“¦w_32px ðŸ“¦just-cnt_center ðŸ“¦box-szg_border-box"
  >
    <svg
      class="ðŸ“¦w_24px ðŸ“¦h_24px ðŸ“¦box-szg_border-box"
      data-icon="menu"
      style="fill: #66788A;"
      viewBox="0 0 20 20"
    >
      <path
        d="M1 6h18c.55 0 1-.45 1-1s-.45-1-1-1H1c-.55 0-1 .45-1 1s.45 1 1 1zm18 3H1c-.55 0-1 .45-1 1s.45 1 1 1h18c.55 0 1-.45 1-1s-.45-1-1-1zm0 5H1c-.55 0-1 .45-1 1s.45 1 1 1h18c.55 0 1-.45 1-1s-.45-1-1-1z"
        fill-rule="evenodd"
      />
    </svg>
  </button>
</div>
`;
